### DISCUSSION TITLE
Table connection in many-to-many relation

### DISCUSSION THREAD

**Discussion by VladKopylov** [prisma/prisma#8521] [Q&A]
Hi community.
I'am trying to make connection between tables in many-to-many relations, but I can't find solution for this. In documentation I found only for IMPLICIT many-to-many relations. Could you provide correct solution or give me related links? 

For example, we have such models:

```prisma
model Post {
  id          Int                 @id @default(autoincrement())
  title       String
  categories  CategoriesOnPosts[]
}

model Category {
  id    Int                 @id @default(autoincrement())
  name  String
  posts CategoriesOnPosts[]
}

model CategoriesOnPosts {
  post        Post     @relation(fields: [postId], references: [id])
  postId      Int       
  category    Category @relation(fields: [categoryId], references: [id])
  categoryId  Int      
  assignedAt  DateTime @default(now())
  assignedBy  String

  @@unique([postId, categoryId], name: "post_division_unique")
}
```

One of my failed attempts to do this...
```ts
prisma.post.update({
        where: {
          id: 1,
        },
        data: {
          categories: {
            connect: { id: 2 },
          },
        },
      });
```

**Comment by ryands17** [comment]
@VladKopylov 👋 

We have an article in the docs on how to connect explicit M-N relations: https://www.prisma.io/docs/support/help-articles/working-with-many-to-many-relations.

Let me know if it works 🙂

**Reply by VladKopylov** [reply]
@ryands17 I think this example is not for my case. 
For example data in Post and Categories already exists, but it doesn't connected. How I should Post( title 'Hello world') connect with two Category with id=1 and id=2?
I didn't find examples in documentation

**Reply by ryands17** [reply]
You can use `connect` instead of `create`:

```ts
await prisma.post.create({
  data: {
    title: 'Hello World',
    tags: {
      create: [
        { tag: { connect: { id: 1 } } },
        { tag: { connect: { id: 2 } } },
      ],
    },
  },
})
```

**Reply by VladKopylov** [reply]
@ryands17 thank you for answer. It looks like what I need. But for my case I need to update (not create new note)  Instead of:
await prisma.post.**create** -> await prisma.post.**update**.

Understanding the linking logic for explicit many-to-many realations  - it is difficult for newcomers (I think)

**Reply by ryands17** [reply]
This will work in a similar manner for update as well. You can pass the same structure of `data` as you do in create.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
